sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/Message.java

```diff
@@ -0,0 +1,335 @@
+            "'systemProperties' cannot be null."));
+        this.properties = new HashMap<>();
+    }
+
+    /**
+     * Gets the set of free-form {@link Message} properties which may be used for passing metadata associated with
+     * the {@link Message}  during Service Bus operations. A common use-case for {@code properties()} is to associate
+     * serialization hints for the {@link #getBody()} as an aid to consumers who wish to deserialize the binary data.
+     *
+     * <p><strong>Adding serialization hint using {@code getProperties()}</strong></p>
+     * <p>In the sample, the type of telemetry is indicated by adding an application property with key "messageType".</p>
+     *
+     * @return Application properties associated with this {@link Message}.
+     */
+    public Map<String, Object> getProperties() {
+        return properties;
+    }
+
+    /**
+     * Properties that are populated by Service Bus. As these are populated by the Service Bus, they are
+     * only present on a <b>received</b> {@link Message}.
+     *
+     * @return An encapsulation of all system properties appended by Service Bus into {@link Message}. {@code null} if
+     * the {@link Message} is not received from the Service Bus service.
+     */
+    public Map<String, Object> getSystemProperties() {
+        return systemProperties;
+    }
+
+    /**
+     *
+     * @return Id of the {@link Message}.
+     */
+    public String getMessageId() {
+        return messageId;
+    }
+
+    /**
+     * Sets the message id.
+     * @param messageId of the {@link Message}.
+     */
+    public void setMessageId(String messageId) {
+        this.messageId = messageId;
+    }
+
+    /**
+     *
+     * @return the contentType of the {@link Message}.
+     */
+    public String getContentType() {
+        return contentType;
+    }
+
+    /**
+     * Sets the content type of the {@link Message}.
+     * @param contentType of the message.
+     */
+    public void setContentType(String contentType) {
+        this.contentType = contentType;
+    }
+
+    /**
+     * Gets the actual payload/data wrapped by EventData.
+     *
+     * <p>
+     * If the means for deserializing the raw data is not apparent to consumers, a common technique is to make use of
+     * {@link #getProperties()} when creating the event, to associate serialization hints as an aid to consumers who
+     * wish to deserialize the binary data.
+     * </p>
+     *
+     * @return A byte array representing the data.
+     */
+    public byte[] getBody() {
+        return Arrays.copyOf(body, body.length);
+    }
+
+    /**
+     * Returns message body as UTF-8 decoded string.
+     *
+     * @return UTF-8 decoded string representation of the event data.
+     */
+    public String getBodyAsString() {
+        return new String(body, UTF_8);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
```

Comment:
Is this equality check correct? I'd assume there are other properties to compare for a Message.

Reply:
Right now, It is comparing object and `Message` body. As I understand more about properties, we can add those checks.